### PR TITLE
feat: added wrapper function for GetStructFromInchI function

### PR DESCRIPTION
**Pull request by qur786**
@maneetgoyal sir , can you look at the GetStructFromINCHI wrapper function , in which the ref( ) is not working with inchi_InputINCHI and inchi_OutputStruct on line no 113 . I don't know why

---

Resolved by adding `@ts-no-check`

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -65,3 +67,49 @@ export interface INCHIOutput {
...
+  for(let opt of options){
+    inputInchiOptions[opt as keyof GetINCHIExOptions] = true;
+  }
+  const optionString = generateOptionsString(inputInchiOptions);
+  const inchiInputObj = {szInChI:"",szOptions: ""};
+  inchiInputObj.szInChI = input;
+  inchiInputObj.szOptions = optionString;
+  const inchiIn = new inchi_InputINCHI(inchiInputObj);
+  const inchiOutStruct = new inchi_OutputStruct();
+  const Output =  INCHIAPI.GetStructFromINCHI(inchiIn.ref(),inchiOutStruct.ref());
```
Comment (maneetgoyal) [inline src/api.ts:113]: Code isnt prettified.

File: src/api.ts [hunk 2]
```diff
@@ -65,3 +67,49 @@ export interface INCHIOutput {
   szMessage: string;
   szLog: string;
 }
+
+export function GetStructFromINCHI(input: string,options: string[] = [""]):0 | -1 | -2 | 1 | 2 | 3 | 4 | 5 {
```
Comment (maneetgoyal) [inline src/api.ts:71]: Declare `0 | -1 | -2 | 1 | 2 | 3 | 4 | 5` as a union type and then reuse it.

Do the same for `0 | -1 | 1 | 2 | 3 | 4` and `0 | -1 | 1 | 2 | 3`.

**Review by maneetgoyal — CHANGES_REQUESTED** [review]
> can you look at the GetStructFromINCHI wrapper function , in which the ref( ) is not working with inchi_InputINCHI and inchi_OutputStruct on line no 113 . I don't know why

Will try to check in a bit

**Comment by qur786** [comment]
@maneetgoyal  sure sir

File: src/api.ts [hunk 3]
```diff
@@ -1,11 +1,17 @@
 import { INCHIAPI } from "./ffis";
+import { inchi_InputINCHI, inchi_OutputStruct } from "./headers";
+import refNAPI from "ref-napi";
 
-export function CheckINCHIKey(input: string): 0 | -1 | 1 | 2 | 3 {
-  return INCHIAPI.CheckINCHIKey(input) as 0 | -1 | 1 | 2 | 3;
+type returnTypeCheckINCHIKey = 0 | -1 | 1 | 2 | 3;
```
Comment (maneetgoyal) [inline src/api.ts:5]: Type should be PascalCased and should all be declared together.

File: src/api.ts [hunk 4]
```diff
@@ -65,3 +71,50 @@ export interface INCHIOutput {
...
+  for (let opt of options) {
+    inputInchiOptions[opt as keyof GetINCHIExOptions] = true;
+  }
+  const optionString = generateOptionsString(inputInchiOptions);
+  const inchiInputObj = { szInChI: "", szOptions: "" };
+  inchiInputObj.szInChI = input;
+  inchiInputObj.szOptions = optionString;
+  const inchiIn = new inchi_InputINCHI(inchiInputObj);
+  const inchiOutStruct = new inchi_OutputStruct();
+  const Output = INCHIAPI.GetStructFromINCHI(inchiIn.ref(), inchiOutStruct.ref());
```
Comment (maneetgoyal) [inline src/api.ts:118]: Variable should be camelCased.

**Review by maneetgoyal — CHANGES_REQUESTED** [review]
Casing issues.

File: src/api.ts [hunk 5]
```diff
@@ -1,11 +1,17 @@
 import { INCHIAPI } from "./ffis";
+import { inchi_InputINCHI, inchi_OutputStruct } from "./headers";
+import refNAPI from "ref-napi";
 
-export function CheckINCHIKey(input: string): 0 | -1 | 1 | 2 | 3 {
-  return INCHIAPI.CheckINCHIKey(input) as 0 | -1 | 1 | 2 | 3;
+type ReturnTypeCheckINCHIKey = 0 | -1 | 1 | 2 | 3;
```
Comment (maneetgoyal) [inline src/api.ts:5]: As mentioned before, all the types should be declared together in a group:

```ts
type ReturnTypeCheckINCHIKey = 0 | -1 | 1 | 2 | 3;
type ReturnTypeCheckINCHI = 0 | -1 | 1 | 2 | 3 | 4;
...
```

**Review by maneetgoyal — CHANGES_REQUESTED** [review]

File: src/api.ts [hunk 6]
```diff
@@ -118,3 +116,93 @@ export function GetStructFromINCHI(input: string, options: string[] = [""]): Sta
...
+    Polymers105: false,
+    NoFrameShift: false,
+    FoldCRU: false,
+    NPZz: false,
+    SAtZZ: false,
+    LargeMolecules: false,
+  };
+  for (let opt of options) {
+    inputInchiOptions[opt as keyof GetINCHIExOptions] = true;
+  }
```
Comment (maneetgoyal) [inline src/api.ts:154]: Create a function to modularize this logic.

File: src/api.ts [hunk 7]
```diff
@@ -118,3 +116,93 @@ export function GetStructFromINCHI(input: string, options: string[] = [""]): Sta
...
+  const optionString = generateOptionsString(inputInchiOptions);
+  const inchiInputObj = { szInChI: "", szOptions: "" };
+  inchiInputObj.szInChI = input;
+  inchiInputObj.szOptions = optionString;
```
Comment (maneetgoyal) [inline src/api.ts:158]:
```suggestion
  const inchiInputObj = { szInChI: input, szOptions: optionString };
```

File: src/api.ts [hunk 8]
```diff
@@ -118,3 +116,93 @@ export function GetStructFromINCHI(input: string, options: string[] = [""]): Sta
...
```
Comment (maneetgoyal) [inline src/api.ts:203]:
```suggestion
  const inchiInputObj = { szInChI: input, szOptions: optionString };
```

File: src/api.ts [hunk 9]
```diff
@@ -118,3 +116,93 @@ export function GetStructFromINCHI(input: string, options: string[] = [""]): Sta
...
+    NoFrameShift: false,
+    FoldCRU: false,
+    NPZz: false,
+    SAtZZ: false,
+    LargeMolecules: false,
+  };
+  for (let opt of options) {
+    inputInchiOptions[opt as keyof GetINCHIExOptions] = true;
+  }
+  const optionString = generateOptionsString(inputInchiOptions);
```
Comment (maneetgoyal) [inline src/api.ts:155]:
```ts
const optionString = generateOptionsString(inputInchiOptions);
```

This line can also go inside the new function.

**Review by maneetgoyal — CHANGES_REQUESTED** [review]

**Comment by qur786** [comment]
sure sir

**Comment by maneetgoyal** [comment]
@qur786 What updates do you have here?

File: src/api.ts [hunk 10]
```diff
@@ -1,19 +1,23 @@
...
 export interface GetINCHIOptions {
+  /**
+   * Meaning: Both ends of wedge point to stereocenters.
+   * Default behavior: Only narrow end of wedge points to stereocenter
+   */
   NEWPSOFF?: boolean;
+  /**
+   * Meaning: All hydrogens in input structure are explicit
+   * Default behavior: Add H according to usual valences
+   */
```
Comment (maneetgoyal) [inline src/api.ts:20]: @qur786 Similar to the other comments, add comments for all the option `keys`.

File: src/api.ts [hunk 11]
```diff
@@ -1,19 +1,58 @@
...
+  /**
+   * Success; no errors or warnings
+   */
+  inchi_Ret_OKAY: 0,
+  /**
+   * Success; warning(s) issued
+   */
+  inchi_Ret_WARNING: 1,
+  /**
+   * Error: no InChI has been created
```
Comment (maneetgoyal) [inline src/api.ts:20]: @qur786 Similar to this object, create objects for `StatusReturnType1` and `StatusReturnType2` too.

**Comment by maneetgoyal** [comment]
Also add unit test cases in a new file (`tests/api.test.ts`) for all these exposed functions.

**Comment by qur786** [comment]
a lot of mistakes on issue history .

**Pull request closed without merging**